src/DataTypes/DataTypeDynamic.cpp

```diff
@@ -71,8 +73,8 @@ static DataTypePtr create(const ASTPtr & arguments)
 
     auto * literal = argument->arguments->children[1]->as<ASTLiteral>();
 
-    if (!literal || literal->value.getType() != Field::Types::UInt64 || literal->value.get<UInt64>() == 0 || literal->value.get<UInt64>() > ColumnVariant::MAX_NESTED_COLUMNS)
-        throw Exception(ErrorCodes::UNEXPECTED_AST_STRUCTURE, "'max_types' argument for Dynamic type should be a positive integer between 1 and 255");
+    if (!literal || literal->value.getType() != Field::Types::UInt64 || literal->value.get<UInt64>() > ColumnVariant::MAX_NESTED_COLUMNS - 1)
+        throw Exception(ErrorCodes::UNEXPECTED_AST_STRUCTURE, "'max_types' argument for Dynamic type should be a positive integer between 0 and 254");
```

Comment:
didn't notice previously, why is 254 hardcoded in string?
You can print `ColumnVariant::MAX_NESTED_COLUMNS - 1`